x/programs/rust/wasmlanche_sdk/src/store/mod.rs

```diff
@@ -6,6 +6,40 @@ use serde::de::DeserializeOwned;
 use serde::{Deserialize, Serialize};
 use serde_bare::{from_slice, to_vec};
 use std::str;
+
+pub enum StoreResult<'a> {
+    Ok(&'a Context),
+    Err(StorageError),
+}
+
+impl StoreResult<'_> {
+    pub fn store_value<T>(self, key: &str, value: &T) -> Self
+    where
+        T: Serialize + ?Sized,
+    {
+        match self {
+            Self::Ok(ctx) => ctx.store_value(key, value),
+            err => err,
+        }
+    }
+    pub fn store_map_value<T, U>(self, map_name: &str, key: &U, value: &T) -> Self
+    where
+        T: Serialize + ?Sized,
+        U: Serialize,
+    {
+        match self {
+            Self::Ok(ctx) => ctx.store_map_value(map_name, key, value),
+            err => err,
+        }
+    }
+    pub fn is_ok(self) -> bool {
+        matches!(self, Self::Ok(_))
+    }
+    pub fn is_err(self) -> bool {
+        matches!(self, Self::Err(_))
+    }
```

Comment:
Yeah, I would convert back to a `Result` when you're ready to do other things like check `is_ok`. For another PR though!